core/model-vocabulary/src/main/java/org/eclipse/rdf4j/model/vocabulary/Vocabularies.java

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+
+package org.eclipse.rdf4j.model.vocabulary;
+
+import org.eclipse.rdf4j.model.IRI;
+import org.eclipse.rdf4j.model.Namespace;
+import org.eclipse.rdf4j.model.base.AbstractIRI;
+import org.eclipse.rdf4j.model.base.AbstractNamespace;
+
+/**
+ * Utility methods related to RDF vocabularies.
+ *
+ * @author Alessandro Bollini
+ * @since 3.5.0
+ *
+ * @implNote To be eventually removed or merged with {@code org.eclipse.rdf4j.model.util.Vocabularies}.
+ */
+class Vocabularies {
+
+	static Namespace createNamespace(String prefix, String namespace) {
+		return new AbstractNamespace() {
+
+			@Override
+			public String getPrefix() {
+				return prefix;
+			}
+
+			@Override
+			public String getName() {
+				return namespace;
+			}
+
+		};
+	}
+
+	static IRI createIRI(final String namespace, final String localName) {
+		return new AbstractIRI() {
```

Comment:
This shift away from using the SimpleValueFactory might impact performance. We'll need to take a closer look at anything involving IRI object comparison.

Reply:
I considered this issue in the design of the current approach and I agree that a closer look is deserved; however, any performance degradation is likely to be traced back to #2546, which factored object equality from `SimpleIRI` to the shared `AbstractIRI` base class.

As far as I can see, the only relevant difference is in the timing of `IRI.toString()`:
failing string object identity, both the original `SimpleIRI.equals()` and the updated `AbstractIRI.equals()` methods compare the return value of `IRI.toString()`, but:

- in the case of `SimpleIRI` the value is precomputed in `AbstractValueFactory` as

```java
public IRI createIRI(String namespace, String localName) {
	return createIRI(namespace + localName);
}
```

- in the case of `AbstractIRI` the value is computed on demand on the basis of the public API as

```java
public String stringValue() {
	return getNamespace() + getLocalName();
}
```

Back to you with some hard figures…